src/main/java/amazin/controller/ShoppingCartRestController.java

```diff
@@ -0,0 +1,85 @@
+package amazin.controller;
+
+import amazin.model.Book;
+import amazin.model.Item;
+import amazin.model.ShoppingCart;
+import amazin.service.ShoppingCartService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.json.simple.JSONObject;
+
+import javax.servlet.http.HttpSession;
+import javax.validation.Valid;
+
+/**
+ * Created by lauramachado on 2019-04-01.
+ */
+
+@RestController
+@SessionAttributes("cart")
+public class ShoppingCartRestController {
+
+    @Autowired
+    ShoppingCartService shoppingCartService;
+
+    @PostMapping("/cart/decrement/{bookId}")
+    public JSONObject decrementItem(@PathVariable("bookId") Long bookId, HttpSession session) {
+        ShoppingCart cart = (ShoppingCart) session.getAttribute("cart");
+        Item item = cart.getItem(bookId);
+
+        if(item.getQuantity() > 1){
```

Comment:
your cart.getItem returns null if the bookId doesn't exist (which can be done by supplying a wrong bookId). then you try to access the item object, this will cause a null pointer exception